src/box/sql/mem.c

```diff
@@ -2641,8 +2641,8 @@ mem_cmp_msgpack(const struct Mem *a, const char **b, int *result,
 		mem.u.u = mp_decode_uint(b);
 		break;
 	case MP_INT:
-		mem.type = MEM_TYPE_INT;
 		mem.u.i = mp_decode_int(b);
+		mem.type = mem.u.i < 0 ? MEM_TYPE_INT : MEM_TYPE_UINT;
```

Comment:
IMO `MEM_TYPE_INT` should be made equivalent to `MP_INT`. Would it be possible?

Reply:
It is possible. I tried to do it at first, but the patch started growing really fast. If you want, I can do that on top, so as not to mix the refactoring with the bugfix.

Should I do it? @locker.

Reply:
OK. If it's too difficult, I don't mind leaving it as is then.